spec.html

```diff
@@ -42240,10 +42533,55 @@ <h1>get Set.prototype.size</h1>
         <emu-alg>
           1. Let _S_ be the *this* value.
           1. Perform ? RequireInternalSlot(_S_, [[SetData]]).
-          1. Let _count_ be 0.
-          1. For each element _e_ of _S_.[[SetData]], do
-            1. If _e_ is not ~empty~, set _count_ to _count_ + 1.
-          1. Return 𝔽(_count_).
+          1. Let _size_ be SetDataSize(_S_.[[SetData]]).
+          1. Return 𝔽(_size_).
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-set.prototype.symmetricdifference">
+        <h1>Set.prototype.symmetricDifference ( _other_ )</h1>
+        <p>This method performs the following steps when called:</p>
+        <emu-alg>
+          1. Let _O_ be the *this* value.
+          1. Perform ? RequireInternalSlot(_O_, [[SetData]]).
+          1. Let _otherRec_ be ? GetSetRecord(_other_).
+          1. Let _keysIter_ be ? GetIteratorFromMethod(_otherRec_.[[Set]], _otherRec_.[[Keys]]).
+          1. Let _resultSetData_ be a copy of _O_.[[SetData]].
+          1. Let _next_ be ~not-started~.
+          1. Repeat, while _next_ is not ~done~,
+            1. Set _next_ to ? IteratorStepValue(_keysIter_).
+            1. If _next_ is not ~done~, then
+              1. If _next_ is *-0*<sub>𝔽</sub>, set _next_ to *+0*<sub>𝔽</sub>.
+              1. Let _resultIndex_ be SetDataIndex(_resultSetData_, _next_).
+              1. If SetDataHas(_O_.[[SetData]], _next_) is *true*, then
+                1. If _resultIndex_ is not ~not-found~, set _resultSetData_[_resultIndex_] to ~empty~.
+              1. Else,
+                1. If _resultIndex_ is ~not-found~, append _next_ to _resultSetData_.
```

Comment:
I find this easier to read if we introduce an `alreadyInResult` alias:

```suggestion
              1. If SetDataIndex(_resultSetData_, _next_) is ~not-found~, let _alreadyInResult_ be *false*. Otherwise let _alreadyInResult_ be *true*.
              1. If SetDataHas(_O_.[[SetData]], _next_) is *true*, then
                1. If _alreadyInResult_ is *true*, set _resultSetData_[_resultIndex_] to ~empty~.
              1. Else,
                1. If _alreadyInResult_ is *false*, append _next_ to _resultSetData_.
```

Reply:
The index is used in one of the substeps, so that doesn't quite work, but I can have both `_resultIndex_` and `_alreadyInResult_`. Not sure that's an improvement really, but whatever.